controllers/machineset_controller_test.go

```diff
@@ -228,6 +228,10 @@ func TestMachineSetReconciler(t *testing.T) {
 			g.Expect(im.GetAnnotations()).To(HaveKeyWithValue("annotation-1", "true"), "have annotations of MachineTemplate applied")
 			g.Expect(im.GetAnnotations()).To(HaveKeyWithValue("precedence", "MachineSet"), "the annotations from the MachineSpec template to overwrite the infrastructure template ones")
 			g.Expect(im.GetLabels()).To(HaveKeyWithValue("label-1", "true"), "have labels of MachineTemplate applied")
+			g.Expect(im.GetOwnerReferences()).To(HaveLen(2))
```

Comment:
I think this should be within an `eventually` block. That's probably what it's making the tests fail.

Reply:
After thinking a little bit on this, I got to the conclusion that this unit test should not verify that the machine controller sets owner ref because this is out of scope (of the machineset_controller_test).
So I have changed to check only the ownerRef that this controller is responsible to set